src/types/program.rs

```diff
@@ -1,21 +1,21 @@
 use crate::{
     serde::deserialize_program::{
-        deserialize_program, Attribute, HintParams, Identifier, InstructionLocation,
+        deserialize_program, Attribute, BuiltinName, HintParams, Identifier, InstructionLocation,
         ReferenceManager,
     },
     types::{errors::program_errors::ProgramError, relocatable::MaybeRelocatable},
 };
 use felt::{Felt, PRIME_STR};
-use serde::Serialize;
+use serde::{Deserialize, Serialize};
 use std::{
     fs::File,
     io::{BufReader, Read},
     {collections::HashMap, path::Path},
 };
 
-#[derive(Clone, Debug, PartialEq, Eq, Serialize)]
+#[derive(Clone, Debug, PartialEq, Eq, Serialize, Deserialize)]
```

Comment:
Is deriving Deserialize necessary here?

Reply:
Well, it's the whole reason this problem was raised.
`Program` is used in starknet_in_rust inside the struct `ContractClass`, which does implement `Deserialize`. So it has to implement it too.
It has been removed in cairo-rs when you changed `Vec<String>` for `Vec<&'static str>`, probably because some errors arise, but I will be needed when starknet_in_rust upgrade its dependency to the last version of cairo-rs

Reply:
Oh, my mistake